src/async.rs

```diff
@@ -0,0 +1,236 @@
+use std::fmt::Arguments;
```

Comment:
Would it be better to have `connection::sync` and `connection::async` modules?

Reply:
As long as we still export `connection::sync::Connection` (and other types from that module) on the root module, we could split it up and not break existing usage.

Reply:
@badboy we never exposed modules anyways. Only the toplevel thing.

Reply:
@mitsuhiko Exactly, that's why I just wanted to mention that we re-export it to keep it working

Reply:
I left it under `src/async.rs` since I would need to re-export it as `async` under the root anyway.